scripts/utils/schemas.py

```diff
@@ -0,0 +1,144 @@
+import os
+from schema import Schema, Optional, Regex, Use, And
+from .constants import PARAMS_FILE, ADD_PATH
+
+
+# leaf key values in the mlcube manifest file can be dicts with a specific structure
+def parse_value(val):
+    if isinstance(val, dict):
+        return val["default"]
+    return val
+
+
+class TrackParamsFile(Schema):
+    """A wrapper around the default schema whose `validate` method
+    accepts a set that keeps track of found optional inputs"""
+
+    def validate(self, data, **kwargs):
+        files = kwargs.pop("files")
+        files.add(PARAMS_FILE)
+        return super().validate(data, **kwargs)
+
+
+class TrackAdditionalFiles(Schema):
+    """A wrapper around the default schema whose `validate` method
+    accepts a set that keeps track of found optional inputs"""
+
+    def validate(self, data, **kwargs):
+        files = kwargs.pop("files")
+        files.add(ADD_PATH)
+        return super().validate(data, **kwargs)
+
+
+# Setup extra inputs validators: params file and additional files
+
+params_validator = And(
+    Use(parse_value),
```

Comment:
Can you for how these small validators work or what they're accomplishing? Just to make it easy to read by someone that doesn't know how schema works (in this case me haha)